### PR TITLE
Set id property on select and textfield/textarea

### PR DESCRIPTION
**Pull request by berenddeboer**
Code which sets the id on the actual input/textarea/select. See #102 for the issue.

### CODE REVIEW

File: src/Internal/Textfield/Implementation.elm [hunk 1]
```diff
@@ -284,6 +284,7 @@ textField lift model options _ =
                 [
                   cs "mdc-text-field__input"
                 , css "outline" "none"
+                , Options.attribute (Html.id index)
```
Comment (fbessou) [inline src/Internal/Textfield/Implementation.elm]: Here are two drawbacks that I think should be taken into account with this implementation:
1 - there is not necessarily only one mdc model in an Elm application so the Index could be used several times in a page and we would have several elements with the same id.
2 - without an explicit warning in documentation there are chances that the user creates an element with the same id as the native control.

**Comment by aforemny** [comment]
Hi @fbessou, thank you for pointing this out. I did implement a semantic change to `Material.index` when migrating to String IDs in [Material.elm](https://github.com/aforemny/elm-mdc/blob/master/src/Material.elm#L199).

**Comment by aforemny** [comment]
Let's continue this discussion in #102.

**Comment by aforemny** [comment]
Hi @berenddeboer, I will close here, because I would like to make this change to all components at once as we have discussed/ will have to discuss in #102.

**Comment by aforemny** [comment]
Hi @berenddeboer, or did you follow up on that here already? I assumed you did not, but now I am not so sure.

**Comment by berenddeboer** [comment]
Haven't done any more work on this, but now I think most of my commits are in, I'm planning to do this along what you suggested, i.e. for all components.

**Comment by aforemny** [comment]
Hi @berenddeboer, I started working on this in 07b4b3cbf9c74179f4a35c8d9d7fcccb011a0390. I ended up implementing it slightly different.

**Comment by aforemny** [comment]
Hi @berenddeboer, I started working on this in 07b4b3cbf9c74179f4a35c8d9d7fcccb011a0390. I ended up implementing it slightly different.

**Pull request closed without merging**